### ISSUE TITLE
Create array of enums

### ISSUE THREAD

**Issue by ovaillancourt** [Automattic/mongoose#661]
Creating an array of enum doesn't appear to be implemented, would be nice to have!

```
var mySchema = new Schema({
  label: [{type:String, enum: ['enumVal1','enumVal2']}]
});
```

Small snippet of conversation on irc which might give direction:

```
aheckmann: file an issue, i think its b/c the validation isn't set on the array
```

**Comment by kwela** [comment]
I use something like
```
roles: { 
    type: [{ type: String,
                enum: [ 'resident', 'distributor', 'company', 'moderator', 'admin' ]
    }],
    default: 'resident'
}
```
to set default value.

**Comment by cloud-tribal** [comment]
What's the status on this ?

**Comment by vkarpov15** [comment]
@cloud-tribal well the issue is closed with 3.9.6 as the milestone, so I'd imagine that means it was released in 3.9.6. If you're using 4.x you should be able to access this feature.
